lib/config.js

```diff
@@ -229,6 +230,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
```

Comment:
Ah good point. The order is deterministic, but yeah, you wouldn't want to bail on `.eslintrc` if `package.json` has `root` or vice-versa.